torch/csrc/autograd/python_variable.cpp

```diff
@@ -199,6 +206,13 @@ PyObject * THPVariable_Wrap(at::TensorBase var)
       status = c10::impl::PyInterpreterStatus::MAYBE_UNINITIALIZED;
     }
   }
+
+  // TODO: is there a better check?
+  if (var.device().type() == c10::kXLA) {
```

Comment:
I suppose you could check the dispatch key, but device is pretty reasonable

Reply:
Also `var.is_xla()` does already exist :)